apps/src/templates/census2017/CensusForm.jsx

```diff
@@ -183,17 +190,43 @@ class CensusForm extends Component {
   }
 
   toggleTopics(option) {
-     const selected = this.state.selectedTopics.slice(0);
-     if (selected.includes(option)) {
-       this.setState({
-         selectedTopics: _.without(selected, option)
-       });
-     } else {
-       this.setState({
-         selectedTopics: selected.concat(option)
-       });
-     }
-   }
+    const selected = this.state.selectedTopics.slice(0);
+    if (selected.includes(option)) {
+      this.clearOption(option);
+    } else {
+      this.selectOption(option);
+    }
+  }
+
+  selectOption(option) {
+    const selected = this.state.selectedTopics.slice(0);
+    this.setState({
+      selectedTopics: selected.concat(option)
+    });
+  }
+
+  clearOption(option) {
+    const selected = this.state.selectedTopics.slice(0);
+    this.setState({
+      selectedTopics: _.without(selected, option)
+    });
+  }
+
+  updateOtherTopicsDesc(event) {
+    const description = event.target.value;
+    const emptyDescription = ("" === description);
+
+    // Clear the "other topics" checkbox when there is no description.
+    if (emptyDescription) {
+      this.clearOption("topic_other_b");
```

Comment:
Does this mean if I select the checkbox, it will immediately clear unless / until I enter some text? I think that might be a bad / confusing user experience. First of all, are we requiring the text, or is it a valid answer to just select other and leave it blank? Obviously if blank is acceptable then this won't work.

Even if an other text is required, I think it would be better to leave the checkbox state alone and validate on submit rather than changing or ignoring user input.

Reply:
Only when the text changes does this function get called. If you don't change the text then the checkbox is left alone. If you select the checkbox you can leave the text blank and the box will remain checked. This does not alter the text based on the checkbox so unchecking won't clear any text already there.

Reply:
Oh, I see. Nice 👍  I misunderstood when this callback was being triggered.